feat(rds): support Database Insights for Aurora databases (#32851)
aws/aws-cdk: packages/aws-cdk-lib/aws-rds/README.md

For more information about Performance Insights, see [Monitoring DB load with Performance Insights on Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.html).

## Database Insights

The standard mode of Database Insights is enabled by default for Aurora databases.

You can enhance the monitoring of your Aurora databases by enabling the advanced mode of Database Insights.

To control Database Insights mode, use the `databaseInsightsMode` property:

```ts
declare const vpc: ec2.Vpc;
new rds.DatabaseCluster(this, 'Database', {
  engine: rds.DatabaseClusterEngine.AURORA,
  vpc: vpc,
  // If you enable the advanced mode of Database Insights,

>   // Performance Insights is enabled and you must set the `performanceInsightRetention` to 465(15 months).
>   databaseInsightsMode: rds.DatabaseInsightsMode.ADVANCED,
>   performanceInsightRetention: rds.PerformanceInsightRetention.MONTHS_15,
>   writer: rds.ClusterInstance.provisioned('Writer', {
>     instanceType: ec2.InstanceType.of(ec2.InstanceClass.R7G, ec2.InstanceSize.LARGE),
>   }),
> });
> ```
> 
> Note: Database Insights are only supported for Amazon Aurora MySQL and Amazon Aurora PostgreSQL clusters.

Reviewer: ```suggestion
  // Performance Insights is enabled and you must set the `performanceInsightRetention` to 465 days (15 months).
  databaseInsightsMode: rds.DatabaseInsightsMode.ADVANCED,
  performanceInsightRetention: rds.PerformanceInsightRetention.MONTHS_15,
  writer: rds.ClusterInstance.provisioned('Writer', {
    instanceType: ec2.InstanceType.of(ec2.InstanceClass.R7G, ec2.InstanceSize.LARGE),
  }),
});
```

**Note**: Database Insights are only supported for Amazon Aurora MySQL and Amazon Aurora PostgreSQL clusters.
```

For more information about Performance Insights, see [Monitoring DB load with Performance Insights on Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.html).

## Database Insights

> The standard mode of Database Insights is enabled by default for Aurora databases.
> 
> You can enhance the monitoring of your Aurora databases by enabling the advanced mode of Database Insights.
> 
> See [AWS docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Database-Insights.html) for more information about Database Insights.
> 
> The following example shows enabling the advanced mode of Database Insights for a clustered database.
> 
> ```ts
> declare const vpc: ec2.Vpc;
> new rds.DatabaseCluster(this, 'Database', {
>   engine: rds.DatabaseClusterEngine.AURORA,
>   vpc: vpc,
>   // If you enable the advanced mode of Database Insights,
>   // Performance Insights is enabled and the retention period of Performance Insights data is set to 465(15 months).
>   databaseInsightsMode: rds.DatabaseInsightsMode.ADVANCED,
>   writer: rds.ClusterInstance.provisioned('Writer', {
>     instanceType: ec2.InstanceType.of(ec2.InstanceClass.R7G, ec2.InstanceSize.LARGE),
>   }),
> });
> ```
> 
> Note that settings regarding Database Insights are only possible at the cluster level.
> 
> ### Supported Engines
> 
> Database Insights supports Amazon Aurora MySQL and Amazon Aurora PostgreSQL.

Reviewer: ```suggestion
The standard mode of Database Insights is enabled by default for Aurora databases.

You can enhance the monitoring of your Aurora databases by enabling the advanced mode of Database Insights.

To control Database Insights mode, use the `databaseInsightsMode` property:

```ts
declare const vpc: ec2.Vpc;
new rds.DatabaseCluster(this, 'Database', {
  engine: rds.DatabaseClusterEngine.AURORA,
  vpc: vpc,
  // If you enable the advanced mode of Database Insights,
  // Performance Insights is enabled and the retention period of Performance Insights data is set to 465(15 months).
  databaseInsightsMode: rds.DatabaseInsightsMode.ADVANCED,
  writer: rds.ClusterInstance.provisioned('Writer', {
    instanceType: ec2.InstanceType.of(ec2.InstanceClass.R7G, ec2.InstanceSize.LARGE),
  }),
});
```

**Note**: Database Insights are only supported for Amazon Aurora MySQL and Amazon Aurora PostgreSQL clusters.

> Visit [CloudWatch Database Insights](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Database-Insights.html) for more details.

```
Feel free to improve further 👍

Author: https://github.com/aws/aws-cdk/pull/32851/commits/056320bee2f94ca1ad0f1f6e2e3485b074f89700
